src/coreclr/src/jit/compiler.cpp

```diff
@@ -2193,250 +2193,179 @@ void Compiler::compSetProcessor()
-            {
-                opts.setSupportedISA(InstructionSet_AES);
-            }
-
-            if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_PCLMULQDQ) && JitConfig.EnablePCLMULQDQ())
-            {
-                opts.setSupportedISA(InstructionSet_PCLMULQDQ);
-            }
-
-            // We need to additionaly check that COMPlus_EnableSSE3_4 is set, as that
-            // is a prexisting config flag that controls the SSE3+ ISAs
-            if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_SSE3) && JitConfig.EnableSSE3() && JitConfig.EnableSSE3_4())
-            {
-                opts.setSupportedISA(InstructionSet_SSE3);
-
-                if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_SSSE3) && JitConfig.EnableSSSE3())
-                {
-                    opts.setSupportedISA(InstructionSet_SSSE3);
-
-                    if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_SSE41) && JitConfig.EnableSSE41())
-                    {
-                        opts.setSupportedISA(InstructionSet_SSE41);
-#ifdef TARGET_AMD64
-                        opts.setSupportedISA(InstructionSet_SSE41_X64);
-#endif // TARGET_AMD64
-
-                        if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_SSE42) && JitConfig.EnableSSE42())
-                        {
-                            opts.setSupportedISA(InstructionSet_SSE42);
-#ifdef TARGET_AMD64
-                            opts.setSupportedISA(InstructionSet_SSE42_X64);
-#endif // TARGET_AMD64
-
-                            if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_POPCNT) && JitConfig.EnablePOPCNT())
-                            {
-                                opts.setSupportedISA(InstructionSet_POPCNT);
-#ifdef TARGET_AMD64
-                                opts.setSupportedISA(InstructionSet_POPCNT_X64);
-#endif // TARGET_AMD64
-                            }
-
-                            if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_AVX) && JitConfig.EnableAVX())
-                            {
-                                opts.setSupportedISA(InstructionSet_AVX);
-
-                                if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_FMA) && JitConfig.EnableFMA())
-                                {
-                                    opts.setSupportedISA(InstructionSet_FMA);
-                                }
-
-                                if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_AVX2) && JitConfig.EnableAVX2())
-                                {
-                                    opts.setSupportedISA(InstructionSet_AVX2);
-                                }
-                            }
-                        }
-                    }
-                }
-            }
-        }
+        instructionSetFlags.AddInstructionSet(InstructionSet_Vector128);
+        instructionSetFlags.AddInstructionSet(InstructionSet_Vector256);
     }
 
-    if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_LZCNT) && JitConfig.EnableLZCNT())
+    if (!JitConfig.EnableSSE())
```

Comment:
The previous nesting was also mirroring the implementation hierarchy. Is this going to remain correct based on VM checks (and also take into account the `COMPlus_EnableIsa=0` checks)?

Reply:
Yes. That is what the code that calls EnsureInstructionSetFlagsAreValid is doing. It represents the hierarchy of support described in /src/coreclr/src/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt, and means we don't have to duplicate the hierarchy in the VM, the JIT, crossgen2, and wherever else we put this thing.